### PR TITLE
refactor: remove @Deprecated methods

### PR DESCRIPTION
**Pull request by monperrus**
Remove @Deprecated methods and types.

Cool factor: I used Spoon in sniper mode for doing this, see method `removeDeprecatedMethods`. It worked, kudos to the sniper mode!

`.cvsignore` is removed because it crashes the sniper pretty-printer.

### CODE REVIEW

**Comment by monperrus** [comment]
Not so many resources on automated removal of deprecated methods, a serious one is [Drupal 9: Automated Deprecated Code Removal - A Proof of Concept](https://pronovix.com/blog/drupal-9-automated-deprecated-code-removal-proof-concept)

**Comment by monperrus** [comment]
ready to merge

File: src/main/java/spoon/reflect/cu/CompilationUnit.java [hunk 1]
```diff
@@ -12,37 +12,4 @@
  * public type declaration and other secondary types declarations (not public).
  */
 public interface CompilationUnit extends CtCompilationUnit {
```
Comment (surli) [inline src/main/java/spoon/reflect/cu/CompilationUnit.java]: I guess the whole interface should be put as deprecated now? or is it still used internally?

File: src/test/java/spoon/test/imports/ImportScannerTest.java [hunk 2]
```diff
@@ -231,94 +228,6 @@ private boolean isTypePresentInStaticImports(String type, Collection<String> sta
 		return imports;
 	}
 
-	@Test
```
Comment (surli) [inline src/test/java/spoon/test/imports/ImportScannerTest.java]: I guess the whole class should be renamed since `ImportScanner` does not exist anymore

Comment (surli) [reply src/test/java/spoon/test/imports/ImportScannerTest.java]: Actually I'm not really comfortable with removing those tests: shouldn't they be refactored?

File: src/test/java/spoon/test/refactoring/RefactoringTest.java [hunk 3]
```diff
@@ -131,4 +132,12 @@ public void testTransformedInstanceofAfterATransformation() {
 		assertEquals("o", instanceofInvocation.getLeftHandOperand().toString());
 		assertEquals("spoon.test.refactoring.testclasses.AClassX", instanceofInvocation.getRightHandOperand().toString());
 	}
-}
+
+	@Test
+	public void testremoveDeprecatedMethods() {
```
Comment (surli) [inline src/test/java/spoon/test/refactoring/RefactoringTest.java]: Would be better to put an ignore on the test, instead of commenting its content, wdyt?

**Comment by monperrus** [comment]
There are testing the implementation, not the functionality, so they can be removed.

**Comment by monperrus** [comment]
I'm not in favor of Ignored tests, this usually results in maintenance and technical debt.

File: src/main/java/spoon/reflect/cu/CompilationUnit.java (hunk 1 above)
Comment (monperrus) [reply src/main/java/spoon/reflect/cu/CompilationUnit.java]: good idea, done.

File: src/test/java/spoon/test/imports/ImportScannerTest.java (hunk 2 above)
Comment (monperrus) [reply src/test/java/spoon/test/imports/ImportScannerTest.java]: you're right the rest if `ImportScannerTest` has been moved to `ImoprtTest`

**Comment by monperrus** [comment]
pushed changes

**Comment by monperrus** [comment]
fixed ci

**Comment by surli** [comment]
> I'm not in favor of Ignored tests, this usually results in maintenance and technical debt.

Well, I'm not a huge fan of commented code, which I find harder to find for maintenance: at least with ignores you're sure that the test is still compiling and you know it's there with the test result.
Now it definitely isn't a blocker for this PR.

Re the refactoring of tests, I didn't check those in the details so I trust you on this one, but I guess we should add a coverage threshold at a point :)

**Comment by monperrus** [comment]
Thanks!

**Pull request merged**